### PR TITLE
Changes to SDK wrappers and functions to allow more flexible scene file processing

### PR DESCRIPTION
**Pull request by amzn-victor**
These changes allow for usage of different asset import SDKs to process scene files. 

- Move AssImp specific code out of node, scene & material wrapper parent classes and into child wrapper classes (AssImpNodeWrapper, etc.), allowing child classes to expose import SDK code. Allows for more convenient implementation of other import SDK's elsewhere (such as in a gem).
- Add a `loadingComponentUuid` parameter to `LoadSceneFromVerifiedPath` to allow for usage of different loading components. Changed tests and all calls to this function accordingly.

### CODE REVIEW

File: Code/Tools/SceneAPI/SDKWrapper/AssImpNodeWrapper.h [hunk 1]
```diff
@@ -20,14 +20,17 @@ namespace AZ
...
+            aiNode* GetAssImpNode() const;
             const char* GetName() const override;
             AZ::u64 GetUniqueId() const override;
             int GetChildCount() const override;
             const std::shared_ptr<NodeWrapper> GetChild(int childIndex) const override;
             const bool ContainsMesh();
             bool ContainsBones(const aiScene& scene) const;
             int GetMaterialCount() const override;
+
+            aiNode* m_assImpNode = nullptr;
```
Comment (amzn-mike) [inline Code/Tools/SceneAPI/SDKWrapper/AssImpNodeWrapper.h:33]: This should probably be marked protected

File: Code/Tools/SceneAPI/SDKWrapper/AssImpSceneWrapper.h [hunk 2]
```diff
@@ -21,15 +21,18 @@ namespace AZ
...
 
             bool LoadSceneFromFile(const char* fileName) override;
             bool LoadSceneFromFile(const AZStd::string& fileName) override;
 
             const std::shared_ptr<SDKNode::NodeWrapper> GetRootNode() const override;
             std::shared_ptr<SDKNode::NodeWrapper> GetRootNode() override;
+            virtual const aiScene* GetAssImpScene() const;
             void Clear() override;
 
+            const aiScene* m_assImpScene = nullptr;
```
Comment (amzn-mike) [inline Code/Tools/SceneAPI/SDKWrapper/AssImpSceneWrapper.h:34]: Same here, protected

**Review by amzn-mike — APPROVED** [review]

File: Code/Tools/SceneAPI/SDKWrapper/AssImpMaterialWrapper.h [hunk 3]
```diff
@@ -38,6 +39,8 @@ namespace AZ
             AZStd::optional<bool> GetUseEmissiveMap() const;
             AZStd::optional<float> GetEmissiveIntensity() const;
             AZStd::optional<bool> GetUseAOMap() const;
+
```
Comment (jackalbe) [inline Code/Tools/SceneAPI/SDKWrapper/AssImpMaterialWrapper.h:42]: should this be protected or private?

File: Code/Tools/SceneAPI/SceneBuilder/SceneImporter.cpp [hunk 4]
```diff
@@ -222,7 +222,12 @@ namespace AZ
                     int childCount = node.m_node->GetChildCount();
                     for (int i = 0; i < childCount; ++i)
                     {
-                        std::shared_ptr<AssImpSDKWrapper::AssImpNodeWrapper> child = std::make_shared<AssImpSDKWrapper::AssImpNodeWrapper>(node.m_node->GetChild(i)->GetAssImpNode());
+                        const std::shared_ptr<SDKNode::NodeWrapper> nodeWrapper = node.m_node->GetChild(i);
+                        auto assImpNodeWrapper = std::dynamic_pointer_cast<AssImpSDKWrapper::AssImpNodeWrapper>(nodeWrapper);
```
Comment (jackalbe) [inline Code/Tools/SceneAPI/SceneBuilder/SceneImporter.cpp:226]: do we really use dynamic_pointer_cast<> in the code normally? Normally we steer away from using the STD dynamic_cast stuff in favor of azrtti_cast<> if possible.

**Review by jackalbe — APPROVED** [review]

File: Code/Tools/SceneAPI/SceneCore/Events/AssetImportRequest.h [hunk 5]
```diff
@@ -102,8 +102,9 @@ namespace AZ
                 //! @param sourceGuid The guid assigned to the source file (not the manifest).
                 //! @param requester The application making the request to load the file. This can be used to optimize the type and amount of data
                 //! to load.
+                //! @param loadingComponentUuid The UUID assigned to the loading component.
                 static AZStd::shared_ptr<Containers::Scene> LoadSceneFromVerifiedPath(const AZStd::string& assetFilePath,
-                    const Uuid&sourceGuid, RequestingApplication requester);
+                    const Uuid&sourceGuid, RequestingApplication requester, const Uuid&loadingComponentUuid);
```
Comment (AMZN-stankowi) [inline Code/Tools/SceneAPI/SceneCore/Events/AssetImportRequest.h:107]: add a space after the & before loadingComponentUuid

**Pull request merged**